src/fmu/ensemble/ensemble.py

```diff
@@ -263,21 +290,43 @@ def add_from_runpathfile(self, runpath, runpathfilter=None, batch=None):
             ):
                 raise ValueError("runpath dataframe not correct")
 
-        for _, row in runpath_df.iterrows():
-            if runpathfilter and runpathfilter not in row["runpath"]:
-                continue
-            logger.info("Adding realization from %s", row["runpath"])
-            realization = ScratchRealization(
-                row["runpath"],
-                index=int(row["index"]),
-                autodiscovery=False,
-                batch=batch,
+        if runpathfilter:
+            runpath_df = runpath_df[runpath_df["runpath"].str.contains(runpathfilter)]
+
+        if use_concurrent():
+            logger.info(
+                "Loading %s realizations concurrently from runpathfile",
+                str(len(runpath_df)),
             )
-            # Use the ECLBASE from the runpath file to
-            # ensure we recognize the correct UNSMRY file
-            realization.find_files(row["eclbase"] + ".DATA")
-            realization.find_files(row["eclbase"] + ".UNSMRY")
-            self.realizations[int(row["index"])] = realization
+            with ProcessPoolExecutor() as executor:
+                loaded_reals = [
+                    executor.submit(
+                        ScratchRealization,
+                        row.runpath,
+                        index=int(row.index),
+                        autodiscovery=False,
+                        find_files=[row.eclbase + ".DATA", row.eclbase + ".UNSMRY",],
+                        batch=batch,
+                    ).result()
+                    for row in runpath_df.itertuples()
+                ]
```

Comment:
Same as above
```suggestion
            with ProcessPoolExecutor() as executor:
                reals_futures = [
                    executor.submit(
                        ScratchRealization,
                        row.runpath,
                        index=int(row.index),
                        autodiscovery=False,
                        find_files=[row.eclbase + ".DATA", row.eclbase + ".UNSMRY",],
                        batch=batch,
                    )
                    for row in runpath_df.itertuples()
                ]
                ```suggestion
               reals_futures = [
                    executor.submit(
                        ScratchRealization,
                        realdir,
                        realidxregexp=realidxregexp,
                        autodiscovery=autodiscovery,
                        batch=batch,
                    )
                    for realdir in globbedpaths
                ]
                loaded_reals = [x.result() for x in reals_futures]
```
```